scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java

```diff
@@ -676,6 +679,22 @@ boolean pauseJob(String jobId) throws NotConnectedException, UnknownJobException
             PermissionException;
 
     /**
+     * Restart all in error tasks in the job represented by jobId.<br>
+     * This method will restart every in error tasks of this job.<br><br>
+     * The jobId is given as a string. It's in fact the string returned by the {@link JobId#value()} method.<br>
+     * A user can only restart HIS job.<br>
+     * If the job does not exist, a schedulerException is sent with the proper message.
+     *
+     * @param jobId the job to resume.
+     * @return true if success, false if not.
+     * @throws NotConnectedException if you are not authenticated.
+     * @throws UnknownJobException if the job does not exist.
+     * @throws PermissionException if you can't access to this particular job.
+     */
+    boolean restartAllInErrorTasks(String jobId) throws NotConnectedException, UnknownJobException,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: org.ow2.proactive.scheduler.common.exception.NotConnectedException, org.ow2.proactive.scheduler.common.exception.UnknownJobException, org.ow2.proactive.scheduler.common.exception.PermissionException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)